### PR TITLE
Manual sync of 1.12 with develop

### PR DESCRIPTION
**Pull request by derobins**
Brings all features from develop. Note that RELEASE.txt has not been
updated (will be done in a future PR).

### CODE REVIEW

**Comment by derobins** [comment]
develop and 1.12 are identical aside from:

* HDFS VFD stubs
* Version numbers
* RELEASE.txt

File: CMakeLists.txt [hunk 1]
```diff
@@ -225,6 +225,7 @@ set (HDF5_JAVA_LIB_DIR             ${HDF5_SOURCE_DIR}/java/lib)
 set (HDF5_JAVA_LOGGING_JAR         ${HDF5_SOURCE_DIR}/java/lib/slf4j-api-1.7.25.jar)
 set (HDF5_JAVA_LOGGING_NOP_JAR     ${HDF5_SOURCE_DIR}/java/lib/ext/slf4j-nop-1.7.25.jar)
 set (HDF5_JAVA_LOGGING_SIMPLE_JAR  ${HDF5_SOURCE_DIR}/java/lib/ext/slf4j-simple-1.7.25.jar)
+set (HDF5_DOXYGEN_DIR              ${HDF5_SOURCE_DIR}/doxygen)
```
Comment (byrnHDF) [inline CMakeLists.txt]: doxygen changes are in #89.

File: config/cmake_ext_mod/HDFLibMacros.cmake [hunk 2]
```diff
@@ -84,6 +84,10 @@ endmacro ()
 #-------------------------------------------------------------------------------
 macro (EXTERNAL_SZIP_LIBRARY compress_type encoding)
   if (${compress_type} MATCHES "GIT")
+#    FetchContent_Declare (SZIP
```
Comment (byrnHDF) [inline config/cmake_ext_mod/HDFLibMacros.cmake:87]: I would leave these out, until they can be tested.

File: configure.ac [hunk 3]
```diff
@@ -1048,6 +1048,64 @@ if test "X$HDF5_TOOLS" = "Xno"; then
...
+if test "X$HDF5_DOXYGEN" = "Xyes"; then
+  DX_DOXYGEN_FEATURE(ON)
+  DX_DOT_FEATURE(OFF)
+  DX_HTML_FEATURE(ON)
+  DX_CHM_FEATURE(OFF)
+  DX_CHI_FEATURE(OFF)
+  DX_MAN_FEATURE(OFF)
+  DX_RTF_FEATURE(OFF)
+  DX_XML_FEATURE(OFF)
+  DX_PDF_FEATURE(ON)
```
Comment (byrnHDF) [inline configure.ac]: #89 defaults to OFF

File: src/Makefile.am [hunk 4]
```diff
@@ -196,12 +198,20 @@ $(top_srcdir)/src/H5version.h: $(top_srcdir)/src/H5vers.txt
 $(top_srcdir)/src/H5overflow.h: $(top_srcdir)/src/H5overflow.txt
 	perl $(top_srcdir)/bin/make_overflow $?
 
+# doxygen support
```
Comment (byrnHDF) [inline src/Makefile.am]: #89 changed how this is done

**Review by byrnHDF — COMMENTED** [review]
I would be okay after #89 is merged

**Comment by derobins** [comment]
> I would be okay after #89 is merged

I just brought it up to speed with what it currently in develop. We can either update this PR with the changes from #89 or we can do that as a separate merge.

**Comment by lrknox** [comment]
I merged #89.
This PR shows merge conflicts in 4 files.  GitHub provides an editor to resolve conflicts;  nice of them.  Resolution for the conflicts in MANIFEST and configure.ac was straightforward and obvious.
m4/ax_prog_doxygen.m4 is version 24, and Allen downgraded it to version 23 in develop, probably after you started this sync.  It should be downgraded (from develop?).
One conflict in src/Makefile.am involved tabs changed to spaces in libhdf5_la_SOURCES.  I kept the spaces, but develop still has the tabs.  The other conflict was a change to an if BUILD_DOXYGEN_CONDITIONAL in current develop, which I took.

I'm not sure if the 3 resolved conflicts are propagated in to the PR or not.  The ax_prog_doxygen.m4 still needs to be done.

File: src/H5Ppublic.h [hunk 5]
```diff
@@ -361,6 +361,8 @@ H5_DLL herr_t      H5Pget_mdc_log_options(hid_t plist_id, hbool_t *is_enabled, c
                                           size_t *location_size, hbool_t *start_on_access);
 H5_DLL herr_t      H5Pset_evict_on_close(hid_t fapl_id, hbool_t evict_on_close);
 H5_DLL herr_t      H5Pget_evict_on_close(hid_t fapl_id, hbool_t *evict_on_close);
+H5_DLL herr_t      H5Pset_file_locking(hid_t fapl_id, hbool_t use_file_locking, hbool_t ignore_when_disabled);
+H5_DLL herr_t H5Pget_file_locking(hid_t fapl_id, hbool_t *use_file_locking, hbool_t *ignore_when_disabled);
```
Comment (lrknox) [inline src/H5Ppublic.h:365]: Spacing seems odd here.  Not critical, but curious as to why.

**Review by lrknox — APPROVED** [review]

File: src/H5Ppublic.h (hunk 5 above)
Comment (derobins) [reply src/H5Ppublic.h:365]: I didn't re-run the formatter since I was moving code from formatted source to formatted source.

**Pull request merged**